internal/gatewayapi/route.go

```diff
@@ -390,16 +390,23 @@ func (t *Translator) processGRPCRouteRule(grpcRoute *GRPCRouteContext, ruleIdx i
 		}
 
 		if match.Method != nil {
-			if match.Method.Method != nil {
+			// GRPC's path is in the form of "/<service>/<method>"
+			// TODO: support regex match type after https://github.com/kubernetes-sigs/gateway-api/issues/1746 is resolved
+			switch {
+			case match.Method.Service != nil && match.Method.Method != nil:
+				irRoute.PathMatch = &ir.StringMatch{
+					Exact: StringPtr(fmt.Sprintf("/%s/%s", *match.Method.Service, *match.Method.Method)),
+				}
+			case match.Method.Method != nil:
 				irRoute.HeaderMatches = append(irRoute.HeaderMatches, &ir.StringMatch{
-					Name:  ":method",
-					Exact: match.Method.Method,
+					Name:   ":path",
```

Comment:
why not use `irRoute.PathMatch` here ?

Reply:
`irRoute.PathMatch` doesn't support `Suffix` match

Reply:
sgtm can you add a comment in here mentioning ^ ?
cc @lizan

Reply:
@arkodg done